nautobot_golden_config/models.py

```diff
@@ -129,22 +147,56 @@ def _verify_get_custom_compliance_data(compliance_details):
             raise ValidationError(VALIDATION_MSG.format(val, "String or Json", compliance_details[val]))
 
 
+def _get_hierconfig_remediation(obj):
+    """Returns the remediation plan."""
+    hierconfig_os = HIERCONFIG_LIB_MAPPER_REVERSE.get(obj.device.platform.slug)
+    if not hierconfig_os:
+        raise ValidationError(f"platform {obj.device.platform.slug} is not supported by hierconfig.")
+
+    try:
+        remediation_setting_obj = RemediationSetting.objects.get(platform=obj.rule.platform)
+    except Exception as err:  # pylint: disable=broad-except:
+        raise ValidationError(f"Platform {obj.device.platform.slug} has no Remediation Settings defined.") from err
+
+    remediation_options = remediation_setting_obj.remediation_options or None
+
+    try:
+        hc_kwargs = {"hostname": obj.device.name, "os": hierconfig_os}
+        if remediation_options:
+            hc_kwargs.update(hconfig_options=remediation_options)
+        host = HierConfigHost(**hc_kwargs)
+
+    except Exception as err:  # pylint: disable=broad-except:
```

Comment:
Are there any specific exceptions we could already handle ?

Reply:
HIer uses builtin exceptions, mainly fails with AttributeError and KeyError, there are no Hier-specific exceptions to catch.